src/TransferList/TransferList.types.ts

```diff
@@ -1,14 +1,79 @@
-export type TransferListProps = {
+export type TransferListItem =
+  | string
+  | { primaryLabel: string; secondaryLabel?: string; id: string };
+
+export type TransferListProps<T> = {
+  /**
+   * The key to be used for the filter.
+   */
+  filterKey?: (item: T) => string;
```

Comment:
What is this for? I can't edit it in any of the stories.

Reply:
This is to indicate the key to be filtering on using search but it can go away if we're expecting the user to map the data to the correct format.